README.md

```diff
@@ -16,9 +16,185 @@ An engine for draw a timeline graph:
+
+export default class AxisBody extends Engine.AxisBody {
+    theme = 'theme-name-which-you-want';
+
+    draw() {
+        return super.draw();
+    }
+}
+```
+
+```javascript
+import * as Engine from 'short-night';
+import AxisBody from './AxisBody';
+import AxisMilestone from './AxisMilestone';
+import AxisScale from './AxisScale';
+
+export default class Axis extends Engine.Axis {
+    theme: string = 'theme-name-which-you-want'; 
+
+    bodyConstructor: typeof Engine.AxisBody = AxisBody;
+    milestoneConstructor: typeof Engine.AxisMilestone = AxisMilestone;
+    scaleConstructor: typeof Engine.AxisScale = AxisScale;
+}
+```
+
+### For Creating Event
+
+Create Event Clasess same as below pattern. For EventAxis Extens `Engine.EventAxis`, EventBody Extens `Engine.EventBody`, EventMark Extens `Engine.EventMark`
+
+```javascript
+import * as Engine from 'short-night';
+
+export default class EventAxis extends Engine.EventAxis {
+    
+    theme = 'theme-name-which-you-want';
+
+    draw() {
+        return super.draw();
+    }
+}
+```
+
+```javascript
+import * as Engine from 'short-night';
+import EventBody from './EventBody';
+import EventAxis from './EventAxis';
+import EventMark from './EventMark';
+
+export default class Event extends Engine.Event {
+    theme = 'theme-name-which-you-want';
+
+    bodyConstructor: typeof Engine.EventBody = EventBody;
+    axisConstructor: typeof Engine.EventAxis = EventAxis;
+    markConstructor: typeof Engine.EventMark = EventMark;
+}
+
+```
+
+
+### Create Timeline Class
+
+```javascript
+export default class Timeline extends Engine.Timeline {
+ 
+    theme = 'theme-name-which-you-want';
+
+    constructor() {
+        .
+        .
+        .
+    }
+
+    axisConstructor = Axis;
+    eventConstructor = Event;
+}
+```
+
+
+
+```javascript
+import * as Engine from 'short-night';
+.
+.
+.
+export const draw = Engine.createDraw<Timeline, typeof Timeline>('roshan-demo', Timeline);
+```
+
+```javascript
+(async function () {
+    const timeline = await draw('#app', [
+        {
+            date: '',
+            endDate: '',
+            title: '',
+            description: ''
+        }
+    ]);
+    const data = timeline.export();
+
+    timeline.destroy();
+
+    draw('#app', data)
+
+}());
+```
+
+
+
 # Documents
 
-Wanna using a timeline? click [here](http://short-night.pea3nut.org/example)
+- More info see the Short-Night engine Public API? click [here](https://foxzilla.github.io/short-night/index.html)
+
+- Wanna using a timeline? click [here](http://short-night.pea3nut.org/example)
+
+- Wanna to develop a new timeline? click [here](https://github.com/FoXZilla/short-night/wiki)
+
+
+# Contributor's Guide
+Pull requests are welcome. For major changes, please open an issue first to discuss what you would like to change.
+
+Please make sure to update tests as appropriate.
+
+1. Fork the project on GitHub.
+
+2. Clone the project.
+
+3. perform the below command
+```sh
+npm link
```

Comment:
Can we run `npm link` without any params?

Reply:
yes, because the npm link performs the local linking process with a particular library.